internal/service/alarms/internal/db/repo/alarms_repository.go

```diff
@@ -21,17 +23,21 @@ type AlarmsRepository struct {
 // GetAlarmEventRecordWithUuid grabs a row of alarm_event_record using uuid
 func (ar *AlarmsRepository) GetAlarmEventRecordWithUuid(ctx context.Context, uuid uuid.UUID) ([]models.AlarmEventRecord, error) {
 	// Build sql query
-	query := `
-SELECT alarm_event_record_id, alarm_definition_id, probable_cause_id, alarm_raised_time, 
-       alarm_changed_time, alarm_cleared_time, alarm_acknowledged_time, 
-       alarm_acknowledged, perceived_severity, extensions, resource_type_id
-FROM alarm_event_record 
-WHERE alarm_event_record_id = $1
-`
+	record := models.AlarmEventRecord{}
+	tags := getAllDBTagsFromStruct(&record)
+
+	query, args, err := psql.Select(
+		sm.Columns(tags.columns()...),
```

Comment:
hmmm probably the original way makes more sense (hardcoding the vals). 

The problem with this that we are sort of forced to  pull all the columns regardless of if we need them. The performance hit (pulling too much useless data + reflection) but worse...degraded readability is probably not worth the tradeoff with simply hardcoding the names of the cols.

Reply:
I have another function where you can specify the name of the fields you want to get. I figured for this single object we could use all of them. For others, we will limit the fields

Reply:
Ah cool. Do you mind specifying the cols? 

If you specify it...there's no point in reflect right?

Reply:
I meant the fields of the struct